packages/gatsby-recipes/src/renderer/render.js

```diff
@@ -117,29 +104,21 @@ const validateResource = (resourceName, context, props) => {
   return error
 }
 
-const handleResource = (
-  resourceName,
-  context,
-  props,
-  allResources,
-  setResources
-) => {
-  const error = validateResource(resourceName, context, props)
-  if (error) {
-    errors.push(error)
-    return null
-  }
-
+const handleResource = (resourceName, context, props) => {
+  // Initialize
   const { mode } = context
 
-  let key
+  const trueKey = props._key ? props._key : context._uuid
+
+  let cacheKey
   // Only run apply once per resource
   if (mode === `apply`) {
-    key = mode + ` ` + resourceName + ` ` + props._key
```

Comment:
some resources have a key & others have auto-generated UUIDs — we weren't reflecting that which meant that all subsequent resources hit the same cache key as the first of its type.